xarray/core/dataarray.py

```diff
@@ -2117,8 +2119,8 @@ def shift(self, shifts=None, **shifts_kwargs):
         Coordinates:
           * x        (x) int64 0 1 2
         """
-        ds = self._to_temp_dataset().shift(shifts=shifts, **shifts_kwargs)
-        return self._from_temp_dataset(ds)
+        return self._replace(variable=self.variable.shift(
+            shifts=shifts, fill_value=fill_value, **shifts_kwargs))
```

Comment:
Is this a reasonable change for `DataArray` operations that only change their data?

Reply:
I think this is 100% equivalent -- I don't really care either way. But perhaps splitting this into two statements would be clearer.